### ISSUE TITLE
EIT rewrite is missing some packets

### ISSUE THREAD

**Issue by doub** [braice/MuMuDVB#24]
tl;dr: EIT rewriting cannot be done easily at the TS packet level.

I've been trying to decode EIT sections from a partial TS streamed by MuMuDVB, and I frequently encounter invalid sections.

I've looked at the code, and it appears that the EIT rewriting code (eit_sort_new_packet) makes two assumptions that are not true in my situation (DVB-T in the UK).

First it assumes that there can be at most only two (partial) sections in a TS packet, ie. one before the pointer_field and one after. It's only checking for the EIT header once at the get_ts_begin offset. But I have TS packets that have several EIT headers (ie. 3 or more section are spanned on the TS packet).

Second, it assumes that if there is a EIT header present, then whatever is before it belongs to the same service. In other words if the TS packet has an EIT header for service X, it won't send the TS packet to another channel. But in my case I have TS packets with an EIT header for service X, but before the header there are some bytes belonging to another EIT section for service Y.

So all in all, the EIT rewriting needs much more work. Packets should be split, with the beginning of a packet sent to whatever channel the previous packet ended with. The end of such a packet should be re-stuffed before being sent to the first channel (and the payload_unit_start_indicator flag cleared). Then each EIT section (and not only the first one) should be independently forwarded to the right channel, which for the second and subsequent EIT headers implies rewriting the pointer_field of the TS packet.

Maybe a simpler solution would be to do the rewriting at the section level, ie. parse the TS packets and drop them, and completely recreate them once a section is completed (which may happen several times during the processing of a single packet).
